scripts/semconv/templates/SemanticAttributes.ts.j2

```diff
@@ -24,9 +37,74 @@
 {%- macro lowerFirst(text) -%}
   {{ text[0]|lower}}{{text[1:] }}
 {%- endmacro %}
+{%- macro format_value(type, value, depth = 0, orgValue = value) -%}
+  {%- if type == "string" %}
+    {%- set ns = namespace(done=false) %}
+    {%- for val in dotConstants %}
+      {%- if value.startswith(val ~ ".") %}
+        {{val|upper}}_DOT + {{ format_value(type, value[((val|length)+1):], depth + 1, orgValue) }}
+        {%- set ns.done = true %}
+      {%- endif %}
+    {%- endfor %}
+    {%- if not ns.done %}
+      {%- for val in constants %}
+        {%- if value == val %}
+          {{val|upper}} as '{{orgValue}}'
+          {%- set ns.done = true %}
+        {%- elif value.startswith(val) %}
+          {{val|upper}} + {{ format_value(type, value[(val|length):], depth + 1, orgValue) }}
+          {%- set ns.done = true %}
+        {%- endif %}
+      {%- endfor %}
+    {%- endif %}
+    {%- if not ns.done %}
+      {%- if value.startswith("_") %}
+        UNDERSCORE + {{ format_value(type, value[1:], depth + 1, orgValue) }}
+        {%- set ns.done = true %}
+      {%- elif value.startswith(".") %}
+        DOT + {{ format_value(type, value[1:], depth + 1, orgValue) }}
+        {%- set ns.done = true %}
+      {%- endif %}
+    {%- endif %}
+    {%- if not ns.done %}
+      '{{value}}'{% if depth > 0 %} as '{{orgValue}}'{% endif %}
```

Comment:
An extension of this and line 72 could be a right to left parse (replacing from the end of the string), there are only a few (2) cases where this would aid in string reuse of the constants, so I stopped short of that. Later versions of the Semantic conventions may make it more viable to spend the time, but just adding new constants for left-to-right parsing is probably easier.